app/controllers/orders_controller.rb

```diff
@@ -316,7 +316,8 @@ def grouped_order_details
     # Do not do any additional scopes on `@order.order_details` to ensure the list
     # of OrderDetail objects is memoized. Validations are run on `@order.order_details`
     # and we want to be able to render validation errors inline.
-    @order.order_details.sort_by(&:group_id).slice_when do |before, after|
+    sorted_order_details = @order.order_details.sort_by { |od| od.group_id.to_i }
```

Comment:
Is it worth taking care of this at the model level, either with a `safe_group_id` method, or by making the group_id have a default?